File: backend/src/main/java/br/ufpr/sept/so2/modules/solicitacoes/api/dto/SolicitacaoEventoResponse.java
package br.ufpr.sept.so2.modules.solicitacoes.api.dto;

import br.ufpr.sept.so2.modules.solicitacoes.domain.SolicitacaoEvento;

import java.time.OffsetDateTime;
import java.util.UUID;

public record SolicitacaoEventoResponse(
        UUID id,
        String tipo,
        String estadoDe,
        String estadoPara,
        UUID atorId,
        String parecer,
        OffsetDateTime createdAt
) {

    public static SolicitacaoEventoResponse from(SolicitacaoEvento evento) {
        return new SolicitacaoEventoResponse(
                evento.id(),
                evento.tipo(),
                evento.estadoDe(),
                evento.estadoPara(),
                evento.atorId(),
                evento.parecer(),
                evento.createdAt()
        );
    }
}

File: backend/src/main/kotlin/br/ufpr/sept/so2/modules/solicitacoes/api/dto/SolicitacaoEventoResponse.kt
package br.ufpr.sept.so2.modules.solicitacoes.api.dto

import br.ufpr.sept.so2.modules.solicitacoes.domain.SolicitacaoEvento
import java.time.OffsetDateTime
import java.util.UUID

data class SolicitacaoEventoResponse(
    val id: UUID,
    val tipo: String,
    val estadoDe: String?,
    val estadoPara: String,
    val atorId: UUID?,
    val parecer: String?,
    val createdAt: OffsetDateTime,
) {
    companion object {
        @JvmStatic
        fun from(evento: SolicitacaoEvento): SolicitacaoEventoResponse =
            SolicitacaoEventoResponse(
                evento.id,
                evento.tipo,
                evento.estadoDe,
                evento.estadoPara,
                evento.atorId,
                evento.parecer,
                evento.createdAt,
            )
    }
}
